script.js

```diff
@@ -1,4 +1,184 @@
+/*Base
+Requirements
+There are 2 players and players take turns.
+When a player clicks Submit, the game rolls 2 dice and shows the dice rolls, for example 3 and 6.
+The player picks the order of the dice they want. For example, if they wanted the number 63, they would specify that the 2nd dice goes first. You can choose how the player specifies dice order.
+After both players have rolled and chosen dice order, the player with the higher combined number wins.*/
+
+// -1: Introduction and game mode selection
+// 0 : Before player 1 roll dice
+// 1 : After player 1 roll dice
+// 2 : Before player 2 roll dice
+// 3 : After player 2 roll dice
+var turnOrOrder = -1;
+
+// 1: Normal mode
+// 2: Reversed Mode
+var gameMode = 0;
+
+// 0 : No action needed
+// 1 : Action needed
+var whichDiceOrder = 0;
```

Comment:
Good use of these three variables and numbers to store state information about your game, but truth be told it was quite hard to follow the code when reading through and I had to repeatedly refer back here to know what each number means. 

For `gameMode` and `whichDiceOrder`, since they can only take two possible values, consider using a boolean for them. So let's say `gameMode`, it can be called `isReverseMode` and set to be `false` at first. Then subsequently in your code where you check say, `if (gameMode == 1)` you can do `if (isReverseMode)` and similarly for normal mode. You can do the same for `whichDiceOrder`, a better name would prob be `isActionNeeded` which is also set to `false` at first. 

This makes your code more readable - the variable name is clearer and the use of the boolean variable means that the code can read more like English. This helps especially when working in large teams where many developers work together. Ensuring that your code is clean and readable for not just for yourself but for your teammates is vital. 

For `turnOrOrder` I get your intention to just have this one variable track all different possible turn states. But considering that this is the key game state which your whole gameplay logic hinges on, I would sacrifice succinctness here for more explicit and elaborate turn states, which again provides the benefit of more readable code in your gameplay below.

```
const GAME_MODE_SELECTION = -1;
const BEFORE_PLAYER_1_ROLL = 0;
const AFTER_PLAYER_1_ROLL = 1;
const BEFORE_PLAYER_2_ROLL = 2;
const AFTER_PLAYER_2_ROLL = 3;
var turnOrOrder = GAME_MODE_SELECTION;
```
Then in your code, you set turnOrOrder to the relevant variable as the game progresses. Then for example in line 40 when you are checking `if ((turnOrOrder == 0 || ... ` it can be `turnOrOrder = BEFORE_PLAYER_1_ROLL` which is much more descriptive. I can instantly know what turn state this is checking for or being updated to without having to reference back up here.